client/components/update-post-status/update-template.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+export default function UpdateTemplate( {
+	post,
+	previousStatus,
+	resetToPreviousState,
+	status,
+	strings: s,
+} ) {
+	let updateClass = 'conf-alert';
+	let updateText, undoTemplate, undoClick, trashText;
+
+	switch ( status ) {
+		case 'trashing':
+		case 'deleting':
+			trashText = ( status === 'deleting' ) ? s.deleting : s.trashing;
+			updateText = (
+				<span>{ trashText } <span className="loading-dot">.</span><span className="loading-dot">.</span><span className="loading-dot">.</span></span>
+			);
+			updateClass += ' conf-alert--trashing';
+			break;
+
+		case 'trash':
+			undoClick = resetToPreviousState;
+			undoTemplate = (
+				<a className="undo" onClick={ undoClick }><span>{ s.undo }</span></a>
+			);
+			updateText = s.trashed;
```

Comment:
Hmm, I suppose we'd need a translated postType for the above example. It might be difficult without context.